python/pyarrow/tests/parquet/test_parquet_file.py

```diff
@@ -277,3 +278,77 @@ def test_pre_buffer(pre_buffer):
     buf.seek(0)
     pf = pq.ParquetFile(buf, pre_buffer=pre_buffer)
     assert pf.read().num_rows == N
+
+
+def test_parquet_file_explicitly_closed(tmpdir):
```

Comment:
```suggestion
def test_parquet_file_explicitly_closed(tempdir):
```

It's probably something we should update (now pytest has a better tmp_path fixture), but at the moment we have an internal fixture `tempdir` that wraps `tmpdir` in a pathlib.Path

Reply:
Ah, that's better! 😄